### PR TITLE
Trac3162

### PR DESCRIPTION
**Pull request by rpetry**
@alf239 - could you review please

### CODE REVIEW

File: scripts/cron/public_private_ae2_to_atlas.sh [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+# Retrieve all Atlas experiments into $all_atlas_experiments_file (json)
+curl -X GET -b $authentication_cookie -H "Accept: application/json" "http://$3:$4/$5/admin?&p=0&n=$num_all_atlas_exps&search=&fromDate=&toDate=&pendingOnly=ALL&op=searchexp&indent" > $all_atlas_experiments_file
+if [ ! -f $all_atlas_experiments_file ]; then
+   err_msg="Updating private/public status of experiments on $3:$4/$5 was unsuccessful due failure to retrieve all Atlas experiments"
+   echo $err_msg >> $process_file.log
+   mailx -s "$err_msg" $6 < $process_file.log
+fi
+
+# Retrieve all AE2 experiments into $all_ae2_experiments_file (ssv)
+curl -X GET "http://www.ebi.ac.uk/arrayexpress/admin/privacy" > $all_ae2_experiments_file
```
Comment (olgamelnichuk) [inline scripts/cron/public_private_ae2_to_atlas.sh:39]: Should the arrayexpress privacy url http://www.ebi.ac.uk/arrayexpress/admin/privacy be working already? curl command returns "The resource you are requesting is not found..." to me.

Comment (rpetry) [inline scripts/cron/public_private_ae2_to_atlas.sh:39]: Sorry Olga, I should have commented with the pull request - the functionality on the AE2 side is not in www yet - but can be tested with wwwdev (wwwdev points at prod AE2 DB also).

File: scripts/cron/public_private_ae2_to_atlas.sh [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+   val=`echo $line | sed 's|[\", ]||g' | grep -P 'private' | cut -d':' -f2`
+   if [ ! -z $val ]; then
+       atlas_private_flag=$val
+
+       if [ ! -z $exp_accession ]; then
+          # Now get the experiment's private/public status in AE2
+          # E.g. line in $all_ae2_experiments_file: accession:E-MEXP-31 privacy:public releasedate:2004-03-01
+          ae2_experiment=`grep "accession:$exp_accession " $all_ae2_experiments_file`
+	      ae2_release_date=`echo $ae2_experiment | awk '{print $3}' | cut -d':' -f2`
+          if [ ! -z ae2_experiment ]; then
```
Comment (olgamelnichuk) [inline scripts/cron/public_private_ae2_to_atlas.sh]: Looks like a typo; ae_experiment variable without "$". Probably it should be like this:

```
if [ ! -z $ae2_experiment ]; then 
   ...
fi
```

Comment (rpetry) [inline scripts/cron/public_private_ae2_to_atlas.sh]: Well spotted - thanks Olga - fixed.

File: scripts/cron/public_private_ae2_to_atlas.sh [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+                   if [ $atlas_private_flag == "false" ]; then
+                      # Experiment private in AE2 and public in Atlas - make it private in Atlas
+                      echo "$exp_accession - AE2: private; Atlas: public - status change in Atlas: public->private" >> $process_file.log
+                      curl -X GET -b $authentication_cookie -H "Accept: application/json" "http://$3:$4/$5/admin?runMode=RESTART&accession=$exp_accession&type=makeexperimentprivate&autoDepends=false&op=schedule"
+                   fi
+               fi
+          else
+             err_msg="Updating private/public status of experiments on $3:$4/$5 unsuccessful: failed to find $exp_accession in AE2"
+             echo $err_msg >> $process_file.log
+             mailx -s "$err_msg" $6 < $process_file.log
```
Comment (olgamelnichuk) [inline scripts/cron/public_private_ae2_to_atlas.sh:88]: Not sure it is good idea to send mail for each failed experiment. In the worst case we get an email per atlas experiment, so about 5 000 emails..

File: scripts/cron/public_private_ae2_to_atlas.sh [hunk 4]
```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+# @author: rpetry
+# @date:   18 Oct 2011
+
+# This script serves to reflect in Atlas private/public experiment status switch on the AE2 side.
+
+if [ $# -eq 0 ]; then
+        echo "Usage: $0 ADMIN_USERNAME ADMIN_PASSWORD ATLAS_URL ATLAS_PORT ATLAS_ROOT ERROR_NOTIFICATION_EMAILADDRESS"
+        echo "e.g. $0 admin <pwd> lime 14032 gxa-load atlas-developers@ebi.ac.uk"
```
Comment (olgamelnichuk) [inline scripts/cron/public_private_ae2_to_atlas.sh]: Do we want to publish email address of our dev mailing list? May be we could use another example here?

Comment (rpetry) [inline scripts/cron/public_private_ae2_to_atlas.sh]: You're right - I'll just put a place holder here instead

File: scripts/cron/public_private_ae2_to_atlas.sh (hunk 3 above)
Comment (rpetry) [inline scripts/cron/public_private_ae2_to_atlas.sh:88]: That is a very good point - I've added exit 1 - the processing should not continue in those cases

Comment (olgamelnichuk) [inline scripts/cron/public_private_ae2_to_atlas.sh:88]: Is it possible to have an experiment in atlas, but not in AE?

Comment (rpetry) [inline scripts/cron/public_private_ae2_to_atlas.sh:88]: in the unlikely event that the file retrieved from AE2 is incomplete - for whatever reason. Not being able to make a decision for a given Atlas experiment due to lack of data from AE2 would be an error situation.

**Pull request merged**